gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -575,7 +529,69 @@ private void addConstructsFinalStateToTaskState(InstrumentedExtractorBase<?, ?>
    * 3. Check whether to publish data in task.
    */
   public void commit() {
-    // TODO- refactor the current Task.run() method to run & commit phases, and add commit content here.
+    try {
+      // Check if all forks succeeded
+      boolean allForksSucceeded = true;
```

Comment:
This check is checking if `fork.commit()` succeeds. The main job of `fork.commit()` is to run `writer.commit()` method to move data from staging to writers' final output. 

The `commit` phase, especially in speculative execution mode, should be something executed only once. So I moved `fork.commit` to the `commit` phase.